nobiru/nobiru_list.py

```diff
@@ -16,3 +16,10 @@ def delete(self, entity):
             db.session.commit()
         except ValueError:
             abort(404, f"{entity._name()} not found")
+
+    def find(self, id):
+        try:
+            return next(x for x in self if x.json()["id"] == id)
```

Comment:
`x.json()["id"] == id`
We shouldn't be serializing an object to extract an attribute for comparison in our code. Serialization methods are the data representation of an object meant for frontend consumption, not for being used as logic in our code. We may not be serializing the id of all objects in the code base, hence this will fail when an object doesn't serialize the id attribute.

Reply:
I hear you.  I'm a little stumped on how to find a note, and verify that it belongs to the ticket, and make it universally applicable.  it doesn't appear as if sqlalchemly has an built in methods to find data within an InstrumentedList.  The only other solution I can think of is to kill this avenue, and do a verification at the endpoint level.  Something akin to:
```python
    def patch(self, ticket_id, id):
        ticket = TicketModel.find(ticket_id)
        note = NotesModel.find(id)

        if note.ticket_id == ticket.id:
            return note.update(schema=NotesSchema, payload=request.json, id=id).json()

        return {"message": "Note not found"}, 404
```

Reply:
well, I think your original implementation of the find method is good. Let's just not serialize it for comparison.
```suggestion
            return next(x for x in self if x.id == id)
```

Does this work?

Reply:
Oh!  That does work.  Sorry I misunderstood. Thanks!

Reply:
Here's the documentation on how to do this with a sql query.
https://docs.sqlalchemy.org/en/14/tutorial/orm_related_objects.html#common-relationship-operators

Reply:
I didn't realize you could do a straight sql query with sqlalchemy.  Do we want to do a query in this instance, or does just using x.id work well enough?